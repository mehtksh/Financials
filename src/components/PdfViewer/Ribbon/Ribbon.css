/* stylelint-disable */
.ribbon {
  display: flex;
  flex-direction: column;
  min-width: 50px;  
  align-items: center;
  border: 1px solid var(--neutral-gray-10);
  border-top: 0px;
}

.ribbon-identifier {
  bottom: 71%;
  right: 1px;
  box-shadow: none;
}

.ribbon-fde {
  display: flex;
  flex-direction: row;
  position: absolute;
  bottom: 8%;
  right: 8px;
  margin-top: -50px;
  z-index: 100;
  min-width: 47px;
  background: white 0% 0% no-repeat padding-box;
  border: 1px solid #dedfe0;
  opacity: 1;
  border-radius: 4px;
  box-shadow: 0 1px 12px rgb(0 0 0 / 8%);
  height: 32px;
}

.ribbon-modal {
  bottom: 42%;
  right: 0.5%;
}

.ribbon-item {
  border: 1px solid var(--primary-blue-78);
  min-height: 2em;
  padding: 9px 9px;
  cursor: pointer;
  display: flex;
  align-items: center;
  justify-content: center;
  border-radius: 4px;
}

.ribbon-item:hover {
  background-color: #f5f5f5;
}

.ribbon-zoomin {
  padding-top: 1rem;
  background: var(--neutral-gray-02)
}

.ribbon-icons {
  margin-top: 8px;
}
