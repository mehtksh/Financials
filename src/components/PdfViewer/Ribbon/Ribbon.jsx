import React, { Component } from "react";
import PropTypes from "prop-types";
import "../Ribbon/Ribbon.css";
import { Tooltip } from "@maknowledgeservices/neptune";
import { FaExpand, FaSearchMinus, FaSearchPlus } from "react-icons/fa";
import { FiRotateCcw } from "react-icons/fi";
class Ribbon extends Component {
  render = () => {
    let { ribbonClass } = this.props;
    return (
      <div className={`${ribbonClass} ${"ribbon-zoomin"}`}>
        <Tooltip
          tip="Zoom-In"
          position="top"
          trigger="hover"
          style={{
            backgroundColor: "var(--neutral-gray-05)",
            color: "var(--neutral-gray-60)",
            whiteSpace: "nowrap",
          }}
        >
          <div className="ribbon-item item-1" onClick={this.props.zoomInView}>
            <FaSearchPlus
              style={{ color: "var(--primary-blue-78)" }}
              data-testid="zoomin"
            />
          </div>
        </Tooltip>
        <Tooltip
          tip="Zoom-Out"
          position="top"
          trigger="hover"
          style={{
            backgroundColor: "var(--neutral-gray-05)",
            color: "var(--neutral-gray-60)",
            whiteSpace: "nowrap",
          }}
        >
          <div
            className="ribbon-item item-3 ribbon-icons"
            onClick={this.props.zoomOutView}
          >
            <FaSearchMinus
              style={{ color: "var(--primary-blue-78)" }}
              data-testid="zoomout"
            />
          </div>
        </Tooltip>
        <Tooltip
          tip="Rotate"
          position="top"
          trigger="hover"
          style={{
            backgroundColor: "var(--neutral-gray-05)",
            color: "var(--neutral-gray-60)",
            whiteSpace: "nowrap",
          }}
        >
          <div
            className="ribbon-item item-2 ribbon-icons"
            style={{ transform: "scaleX(-1)" }}
            onClick={this.props.rotateView}
          >
            <FiRotateCcw style={{ color: "var(--primary-blue-78)" }} />
          </div>
        </Tooltip>
        {!this.props.pdfModalView ? (
          <Tooltip
            tip="Maximize"
            position="top"
            trigger="hover"
            style={{
              backgroundColor: "var(--neutral-gray-05)",
              color: "var(--neutral-gray-60)",
              whiteSpace: "nowrap",
            }}
          >
            {" "}
            <div
              className="ribbon-item item-4 ribbon-icons"
              onClick={this.props.previewView}
            >
              <FaExpand style={{ color: "var(--primary-blue-78)" }} />
            </div>
          </Tooltip>
        ) : (
          // <div className="ribbon-item item-4" onClick={this.props.hideModal}>
          //   <BiExitFullscreen  />
          // </div>
          <></>
        )}
      </div>
    );
  };
}

export default Ribbon;

Ribbon.defaultProps = {
  pdfModalView: false,
  modalVisible: false,
  ribbonClass: false,
  zoomInView: () => {
    // This is intentional
  },
  zoomOutView: () => {
    // This is intentional
  },
  previewView: () => {
    // This is intentional
  },
  hideModal: () => {
    // This is intentional
  },
  rotateView: () => {
    // This is intentional
  },
};

Ribbon.propTypes = {
  pdfModalView: PropTypes.bool,
  modalVisible: PropTypes.bool,
  ribbonClass: PropTypes.bool,
  zoomInView: PropTypes.func,
  zoomOutView: PropTypes.func,
  previewView: PropTypes.func,
  hideModal: PropTypes.func,
  rotateView: PropTypes.func,
};
