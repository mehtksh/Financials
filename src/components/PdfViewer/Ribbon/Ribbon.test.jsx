import renderer from "react-test-renderer";
import React, { Component } from "react";

import { Tooltip } from "@maknowledgeservices/neptune";
import { FaExpand, FaSearchMinus, FaSearchPlus } from "react-icons/fa";
import { FiRotateCcw } from "react-icons/fi";
import Ribbon from "./Ribbon";

jest.mock("../Ribbon/Ribbon.css");
jest.mock("@maknowledgeservices/neptune");
jest.mock("react-icons/fa");
jest.mock("react-icons/fi");

const renderTree = (tree) => renderer.create(tree);
describe("<Ribbon>", () => {
  it("should render component", () => {
    expect(renderTree(<Ribbon />).toJSON()).toMatchSnapshot();
  });
  it("should render component with props", () => {
    expect(
      renderTree(
        <Ribbon
          pdfModalView={false}
          modalVisible={false}
          ribbonClass={false}
          zoomInView={jest.fn()}
          zoomOutView={jest.fn()}
          previewView={jest.fn()}
          hideModal={jest.fn()}
          rotateView={jest.fn()}
        />
      ).toJSON()
    ).toMatchSnapshot();
  });
});
