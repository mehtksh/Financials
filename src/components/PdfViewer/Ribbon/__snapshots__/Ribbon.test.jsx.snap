// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Ribbon> should render component 1`] = `
<div
  className="false ribbon-zoomin"
/>
`;

exports[`<Ribbon> should render component with props 1`] = `
<div
  className="false ribbon-zoomin"
/>
`;
